### ISSUE TITLE
I can build locally but not within Docker

### ISSUE THREAD

**Issue by javierguzman** [getsentry/sentry-webpack-plugin#227]
Hello all,

The time for deploying my first app version has arrived and I am trying to build successfully in production mode. If I run npm run build:production everything goes well and I see sentry plugin working ok (I do not get any build errors).

However, if I want to build a Docker image with the production stuff running this:

`DOCKER_BUILDKIT=1 docker build -f ./config/Dockerfile --target=production -t kindoiserver:production .`

Then first I get the following error:
`
#15 11.13 ERROR in Sentry CLI Plugin: Unable to determine version. Make sure to include `release` option or use the environment that supports auto-detection https://docs.sentry.io/cli/releases/#creating-releases
#15 11.16 npm ERR! code ELIFECYCLE
`
Which I solve adding the release field as follows:
```javascript
    new SentryWebpackPlugin({
      include: outputPath,
      configFile: path.resolve(configPath, 'sentryConfig.properties'),
      ignore: ['node_modules', 'config', '.vscode', 'logs', 'assets'],
      release: process.env.npm_package_version
    }),
```

Then after this I get this other error:

` ERROR in Sentry CLI Plugin: Command failed: /opt/app/server/node_modules/@sentry/cli/sentry-cli releases new 1.0.0
#15 10.83   INFO    2020-09-25 16:04:45.668584209 +00:00 sentry-cli was invoked with the following command line: "/opt/app/server/node_modules/@sentry/cli/sentry-cli" "releases" "new" "1.0.0"
#15 10.83 error: API request failed
#15 10.83   caused by: [60] SSL peer certificate or SSH remote key was not OK`

Why do I not need the release field when building locally in my machine without Docker? What about that SSL? Is this the expected behavior?

Thank you in advance and regards.

**Comment by javierguzman** [comment]
Any feedback about this? Does anyone need more information or something?

**Comment by kamilogorek** [comment]
> Why do I not need the release field when building locally in my machine without Docker?

I don't know what image or what your Dockerfile configuration is, but it's usually caused by `.git` directory, or `git` command itself not being accessible. We are doing a release proposal based on the VCS that the project uses.

> What about that SSL? Is this the expected behavior?

It's definitely not, however it's most likely caused by the docker image missing certificates configuration.
You can verify this by disabling ssl for sentry-cli using `configFile` option in webpack plugin in conjunction with `verify_ssl` config - https://docs.sentry.io/product/cli/configuration/

```
[http]
verify_ssl=false
```

**Comment by javierguzman** [comment]
Thanks @kamilogorek ! From what you say then, it makes sense then... I pull from node image (FROM node:14-slim), which I believe does not come with git and that is why I get the release error within Docker but not locally, if I understood you correctly, right?

I have seen and used the verify_ssl= false option and indeed that got rid of the error. However, is there any implication about setting that option to false?

I guess the certificates is again due to the use of node image.

**Comment by kamilogorek** [comment]
> that is why I get the release error within Docker but not locally, if I understood you correctly, right?

Correct. If `git` would be present, it'd work the same in both environments.

>  is there any implication about setting that option to false?

As long as you control both parties of a transaction, there's not that many, however, it makes it possible to spoof the certificate in the mitm attacks.

> I guess the certificates is again due to the use of node image.

Also correct. Once you configure ssl certificate (self-signed is totally fine), the above error will be gone as well.

**Comment by javierguzman** [comment]
Hello @kamilogorek sorry for the delay but my router passed to a better life.

One last question, when you say  "control both parties of the transaction"? One side is my docker image/environment, but the other one is Sentry, which I do not have any control right?

**Comment by javierguzman** [comment]
Just in case is useful for someone in future I did not know node Docker image was based on Debian so I have got rid of the SSL error doing this within the image build:

RUN apt-get update && apt-get install -y ca-certificates

For the release error one I have ended up playing around with some environment variables rather than relying on Git. I do not fancy copying over to the Docker image the Git folder.

I close this @kamilogorek. Thanks for your help!
